src/CornerstoneViewport/CornerstoneViewport.js

```diff
@@ -281,17 +279,20 @@ class CornerstoneViewport extends Component {
       cornerstoneTools.stopClip(this.element);
     }
 
+    // ~~ OVERLAY
+    if (isOverlayVisible !== prevIsOverlayVisible)
+      updatedState.isOverlayVisible = isOverlayVisible
+
     // ~~ STATE: Update aggregated state changes
     if (Object.keys(updatedState).length > 0) {
       this.setState(updatedState);
     }
 
-    if (isOverlayVisible !== prevIsOverlayVisible)
-      this.setState({ isOverlayVisible });
-
-    // Rebind event listeners when props change to handle functional component method
-    this._bindExternalEventListeners(true);
-    this._bindExternalEventListeners();
+    /* Rebind event listeners when props change to handle functional component method
+    this._bindExternalEventListeners("cornerstone", true);
+    this._bindExternalEventListeners("element", true);
+    this._bindExternalEventListeners("cornerstone");
+    this._bindExternalEventListeners("element"); */
```

Comment:
I think the thought here is that, at time of binding, we use values in props inside the bound methods. If we don't unbind/rebind, we could have the wrong values.